### DISCUSSION TITLE
Apprise Integration into SABnzbd (Do away with the NZB-Notify Addon)

### DISCUSSION THREAD

**Discussion by caronc** [sabnzbd/sabnzbd#2579] [Ideas]
With respect to [nzb-notify/96](https://github.com/caronc/nzb-notify/pull/96) and more importantly [sabnzbd/2549](https://github.com/sabnzbd/sabnzbd/issues/2549), I wanted to illustrate a solution where you could integrate Apprise right into SABnzbd and not lose the functionality you have now. 

Here is a graphic I made:
![Apprise in SABnzbd](https://github.com/sabnzbd/sabnzbd/assets/850374/94e660d8-e460-4174-bc7a-eabaf3b3873c)

**Edit**: One thing to note about the graphic above, is during the instantiation, you only want to add the tags to the URL that the user selected. This is very important. These become the triggers.

I think in the past, there were concerns about having different priorities leveraged with your already built in Pushover calls that were prioritize differently with different states... Hence `on_download` would have a priority of `high`, where an `on_queue` would have a priority of `low` or what not.  This graphic (hence this proposed solution) above easily accommodates for that because the notification service can simply just be added twice, with `?priority=high` set on one, and `?priority=low` set on another.  Then just toggle the tags/categories as required.

Adding new categories or new trigger end points in the future to accomodate other events (such as poor bandwidth performance) is as simple as just associating a new tag with a URLs. After that, just call `notify()` with your new tag you introduced.

At the end of the day the code is more manageable. Your notifications are decoupled and delivered asynchronously too.

Anyway, I love your product. I just wanted to chime in and offer my two cents.

**Comment by thezoggy** [comment]
Awhile back we planned on dropping our built in notifications for external stuff (keeping email/os notifications) and just direct users to use nzb-notify/scripts. In response to that announcement, docker maintainers started to include nzb-notify/apprise where people then could just call that from sab.   

We had to back out of that plan due to limits found with doing priorities:
https://github.com/sabnzbd/sabnzbd/issues/2093#issuecomment-1068404658

From what your are saying above, you'd like us to just actually include apprise with sab (like a module) and just integrate that as the notificaiton page?

**Reply by caronc** [reply]
> Awhile back we planned on dropping our built in notifications for external stuff (keeping email/os notifications) and just direct users to use nzb-notify/scripts. In response to that announcement, docker maintainers started to include nzb-notify/apprise where people then could just call that from sab.
> 
> We had to back out of that plan due to limits found with doing priorities: [#2093 (comment)](https://github.com/sabnzbd/sabnzbd/issues/2093#issuecomment-1068404658)

I remember this 🙂 

> From what your are saying above, you'd like us to just actually include apprise with sab (like a module) and just integrate that as the notificaiton page

Exactly 💯

I just think it would remove a ton of overhead (additional code) for you guys. Not only that, you'd be now doing exactly what the existing nzb-notify script does already (invalidating it) while adding lot more customization and features you couldn't otherwise keep in the addon. Now you can store multiple urls and trigger categories in your own standard config file.

**Reply by caronc** [reply]
I guess we can close this discussion now 🙏👍

**Reply by Safihre** [reply]
Yes, thanks!
